### PR TITLE
Link display currency to account

### PR DESCRIPTION
**Pull request by MaxStalker**
This PR will cover issue #206

### CODE REVIEW

File: src/services/localStorage.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export const localStorageExists =
+  localStorage && typeof localStorage.setItem === "function";
```
Comment (TimDaub) [inline src/services/localStorage.js]: I'm not aware of any browser in the last 10 years that doesn't support `localStorage`: https://caniuse.com/#search=localstorage

I don't think this function is useful.

File: src/components/Advanced.js [hunk 2]
```diff
@@ -25,26 +26,29 @@ export default class Advanced extends React.Component {
...
       expertMode:false
     }
+    this.updateCurrency = this.updateCurrency.bind(this)
   }
 
   componentDidMount() {
-    let currency = localStorage.getItem('currency')
-    const expertMode = localStorage.getItem("expertMode") === "true"
+    const { address } = this.props;
+    const currency = getStoredValue('currency', address) || CURRENCY.DEFAULT_CURRENCY;
```
Comment (TimDaub) [inline src/components/Advanced.js]: Why add `CURRENCY.DEFAULT_CURRENCY` here?

File: src/services/localStorage.js (hunk 1 above)
Comment (MaxStalker) [reply src/services/localStorage.js]: This was originally in the code. I simply moved it out to DRY the code :)
Do you want me to remove this conditional alltogether?

File: src/App.js [hunk 3]
```diff
@@ -1433,15 +1446,15 @@ export default class App extends Component {
                         let upperBoundOfSearch = this.state.block
                         //parse through recent transactions and store in local storage
 
-                        if(localStorage&&typeof localStorage.setItem === "function"){
+                        if(localStorageExists){
```
Comment (TimDaub) [inline src/App.js]: remove check

File: src/App.js [hunk 4]
```diff
@@ -1303,11 +1314,13 @@ export default class App extends Component {
                             if(RNMessageChannel){
                               RNMessageChannel.send("burn")
                             }
-                            if(localStorage&&typeof localStorage.setItem === "function"){
-                              localStorage.setItem(this.state.account+"loadedBlocksTop","")
-                              localStorage.setItem(this.state.account+"metaPrivateKey","")
-                              localStorage.setItem(this.state.account+"recentTxs","")
-                              localStorage.setItem(this.state.account+"transactionsByAddress","")
+                            if(localStorageExists){
```
Comment (TimDaub) [inline src/App.js]: remove check

File: src/App.js [hunk 5]
```diff
@@ -161,8 +162,9 @@ export default class App extends Component {
   // NOTE: This function is for _displaying_ a currency value to a user. It
   // adds a currency unit to the beginning or end of the number!
   currencyDisplay(amount, toParts=false, convert=true) {
-    const locale = localStorage.getItem('i18nextLng') 
-    const symbol = localStorage.getItem('currency');
+    const { account } = this.state;
+    const locale = getStoredValue('i18nextLng');
+    const symbol = getStoredValue('currency', account) || CONFIG.CURRENCY.DEFAULT_CURRENCY;
```
Comment (TimDaub) [inline src/App.js]: Why add `CONFIG.CURRENCY.DEFAULT_CURRENCY`here?

File: src/services/localStorage.js (hunk 1 above)
Comment (TimDaub) [reply src/services/localStorage.js]: > This was originally in the code.

Ah yes, I spotted it only after I saw your function! I think we can safely remove it!

File: src/components/Advanced.js (hunk 2 above)
Comment (MaxStalker) [reply src/components/Advanced.js]: Had a weird bug, where currency was null and would crash the app.
Can't reproduce it now though :\

File: src/App.js (hunk 5 above)
Comment (MaxStalker) [reply src/App.js]: Ah, right - it was actually here :)
If you clear local storage and remove that default value the app will crash

File: src/components/Advanced.js (hunk 2 above)
Comment (TimDaub) [reply src/components/Advanced.js]: OK. We might as well leave it then I guess.

**Comment by MaxStalker** [comment]
> Looks solid so far. I left a couple of comments however.

Thank you for review, Tim!
 
> One major problem I still see with this PR is: We're now "live" with peep.cash, so we'll have to migrate user's "currency" settings to "currency" somehow. If possible, please add that to this PR.

I've added handler that will check if there was a value with name "currency" in localStorage.

**Comment by MaxStalker** [comment]
> Thanks for adjusting the PR.
> 
> Unfortunately, I've spotted a second problem. It is that there's leftovers of the old `localStorage.getItem("currency")` implementation still in the code.
> 
> I think once you've replaced all of those, we should be good to merge.

Hm... can you point me to where you have found those?
I have 0 occuracies found for "localStorage.get" in "src" folder here...

**Pull request merged**